tensordict/utils.py

```diff
@@ -1862,3 +1862,65 @@ def _index_preserve_data_ptr(index):
     if isinstance(index, slice) and (index.start == 0 or index.start is None):
         return True
     return False
+
+
+def remove_duplicates(
+    tensordict: TensorDictBase, key: str, dim: int = 0
+) -> TensorDictBase:
+    """Removes indices duplicated in 'key' along the specified dimension.
+
+    This method detects duplicate elements in the tensor associated with the specified 'key' along the specified
+    'dim' and removes elements in the same indices in all other tensors within the TensorDict.
+
+    Args:
+        tensordict (TensorDictBase): The TensorDict containing potentially duplicate elements.
+        key (str): The key of the tensor along which duplicate elements should be identified and removed.
+        dim (int): The dimension along which duplicate elements should be identified and removed. Default: 0.
```

Comment:
```suggestion
        dim (int): The dimension along which duplicate elements should be identified and removed.
            Defaults to ``0``.
```
Also I guess dim must be within the batch-size right?